test/gtest/ucp/test_ucp_tag_xfer.cc

```diff
@@ -1147,4 +1150,60 @@ UCS_TEST_P(test_ucp_tag_stats, rndv_unexpected, "RNDV_THRESH=1000") {
 
 UCP_INSTANTIATE_TEST_CASE(test_ucp_tag_stats)
 
+class multi_rail_max : public test_ucp_tag_xfer {
+public:
+    void init()
+    {
+        stats_activate();
+        test_ucp_tag_xfer::init();
+    }
+
+    void cleanup()
+    {
+        test_ucp_tag_xfer::cleanup();
+        stats_restore();
+    }
+};
+
+UCS_TEST_P(multi_rail_max, max_lanes, "IB_NUM_PATHS?=8", "TM_SW_RNDV=y")
+{
+    /* MAX_RNDV_LANES must be initialized after 'init' */
+    modify_config("MAX_RNDV_LANES", "8");
+    entity *e1 = create_entity(true);
+    entity *e2 = create_entity(false);
+
+    e1->connect(e2, get_ep_params());
+    e2->connect(e1, get_ep_params());
```

Comment:
test_ucp_tag_xfer already creates connected entities (sender(), receiver()) - can't we use them?
maybe add virtual method num_lanes() [default=1] that test_ucp_tag_xfer::init() will use?